File: packages/core/src/privacy/Redaction.ts
export type RedactionType =
  | "email"
  | "phone"
  | "token"
  | "api_key"
  | "credit_card"
  | "ssn";

export interface Redaction {
  type: RedactionType;
  original: string;
  replacement: string;
  position: {
    start: number;
    end: number;
  };
}

export interface RedactionResult {
  redactedText: string;
  redactions: Redaction[];
  safe: boolean;
}

File: packages/compose/src/main/kotlin/ai/mobileagentos/compose/privacy/Redaction.kt
package ai.mobileagentos.compose.privacy

enum class RedactionType {
    EMAIL,
    PHONE,
    TOKEN,
    API_KEY,
    CREDIT_CARD,
    SSN
}

data class Redaction(
    val type: RedactionType,
    val original: String,
    val replacement: String,
    val position: IntRange
)

data class PrivacyResult(
    val redactedText: String,
    val redactions: List<Redaction>,
    val safe: Boolean
)